clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy

```diff
@@ -219,13 +219,27 @@ class UpsertOpenstackLoadBalancerAtomicOperation extends AbstractOpenstackLoadBa
       task.updateStatus UPSERT_LOADBALANCER_PHASE, "Created listener $name in ${region}"
       task.updateStatus UPSERT_LOADBALANCER_PHASE, "Creating pool $name in ${region}"
       LbPoolV2 lbPool = blockingStatusChecker.execute {
-        provider.createPool(region, name, currentListener.internalProtocol.name(), algorithm.name(), listener.id)
+        provider.createPool(region, name, setInternalProtocol(currentListener.externalProtocol.name()), algorithm.name(), listener.id)
       }
       task.updateStatus UPSERT_LOADBALANCER_PHASE, "Created pool $name in ${region}"
       updateHealthMonitor(region, loadBalancerId, lbPool, healthMonitor)
     }
   }
 
+  private static String setInternalProtocol(String externalProtocol) {
```

Comment:
Might be cleaner to take in the in the enum and switch based upon the enum type like this:

switch(externalProtocol) {
case ListenerProtocol.TCP:
...